src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -3501,6 +3501,10 @@ mono_emit_simd_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 	if (!strcmp (class_ns, "System.Numerics") && !strcmp (class_name, "Vector")){
 		return emit_sri_vector (cfg, cmethod, fsig, args);
 	}
+
+	if (!strcmp (class_ns, "System.Numerics") && !strcmp (class_name, "Vector`1")){
+		return emit_vector64_vector128_t (cfg, cmethod, fsig, args);
+	}
```

Comment:
Aren't these types called Vector64<T> and Vector128<T> ?

Reply:
No, they don't. This is what I observed, without this change:
This is my app https://gist.github.com/fanyang-mono/032126329c3052e8cc2d074af3cf4469
This is the codegen log for System.Numerics.Vector`1<int16>:op_Addition https://gist.github.com/fanyang-mono/4704bdc2b426aae4a2594afddb43e02c
This is the codegen log for System.Runtime.Intrinsics.Vector128`1<int16>:op_Addition https://gist.github.com/fanyang-mono/a9117311d829ba5433f6a11baf9befcb

You can see the add operator of Vector128 got replaced with intrinsics, but not the Vector one.

Reply:
It calls emit_vector64_vector128_t (), are the Vector<T> intrinsics the same as the Vector64<T> etc ones ?

Reply:
On arm64, vector is equivalent to vector128. On amd64, vector is either equivalent to vector128 or vector256.